### PR TITLE
Profile: Edit profile

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

visdiff should show what the new screen looks like -- it's accessed by clicking on any profile field or the Edit Profile button.

### CODE REVIEW

File: shared/profile/edit-profile/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  bio: string,
+  fullname: string,
+  location: string
+}
+
+class EditProfile extends Component<void, Props, State> {
+  props: Props;
+  state: State;
+
+  static parseRoute (currentPath, uri) {
```
Comment (cjb) [inline shared/profile/edit-profile/index.js]: Not sure why I need an empty parseRoute here, when the component's being passed as a subroute of its parent.. any ideas?

File: shared/profile/index.js [hunk 2]
```diff
@@ -59,15 +62,18 @@ export default connect(
     onUserClick: username => { dispatch(routeAppend({path: 'profile', username})) },
     onBack: () => dispatch(navigateUp()),
     onFolderClick: folder => dispatch(openInKBFS(folder.path)),
+    onEditProfile: () => dispatch(routeAppend({path: 'editprofile'})),
   }),
   (stateProps, dispatchProps, ownProps) => {
     const username = ownProps.username || stateProps.myUsername
-    const isYou = username === stateProps.username
+    const isYou = username === stateProps.myUsername
```
Comment (cjb) [inline shared/profile/index.js]: This was a bug in the existing code, failing to recognize the current user.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0f74479f1182...bf3f2e327414 introduced some visual changes on linux:
 * Added: **Edit_Profile-Too_long**  
   Edit_Profile-Too_long rendered
 * Added: **Edit_Profile-Normal**  
   Edit_Profile-Normal rendered

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0f74479f1182...ae19fd2ef4a5 introduced some visual changes on linux:
 * Added: **Edit_Profile-Too_long**  
   Edit_Profile-Too_long rendered
 * Added: **Edit_Profile-Normal**  
   Edit_Profile-Normal rendered

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0f74479f1182...7d415a95ba70 introduced some visual changes on linux:
 * Added: **Edit_Profile-Too_long**  
   Edit_Profile-Too_long rendered
 * Added: **Edit_Profile-Normal**  
   Edit_Profile-Normal rendered

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0f74479f1182...43692a76a667 introduced some visual changes on linux:
 * Added: **Edit_Profile-Too_long**  
   Edit_Profile-Too_long rendered
 * Added: **Edit_Profile-Normal**  
   Edit_Profile-Normal rendered

**Comment by chromakode** [comment]
You can set a `minWidth`/`height` for the dumb component map like this so the screenshots fill more space: https://github.com/keybase/client/blob/5eb676990772c408ef756a2561d4f934c2713f26/shared/profile/dumb.js#L265

File: shared/profile/edit-profile/render.desktop.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            onEnterKeyDown={this.props.onSubmit}
+            onChangeText={location => this.props.onLocationChange(location)} />
+
+          <Input
+            style={styleEditProfile}
+            floatingLabelText='Bio'
+            value={this.props.bio}
+            errorText={this.props.bioLengthLeft <= 5 ? this.props.bioLengthLeft + ' characters left.' : ''}
+            onEnterKeyDown={this.props.onSubmit}
+            onChangeText={bio => this.props.onBioChange(bio)} />
```
Comment (awendland) [inline shared/profile/edit-profile/render.desktop.js]: This input is multiline in the design, and has a smaller font than the other inputs

screen shot 2016-07-20 at 10 14 48 am

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 0f74479f1182...6a18cebb0426 introduced some visual changes on linux:
 * Added: **Edit_Profile-Too_long**  
   Edit_Profile-Too_long rendered
 * Added: **Edit_Profile-Normal**  
   Edit_Profile-Normal rendered

**Comment by cjb** [comment]
@awendland @chromakode Thanks!  I made the inputs and the dumb pane wider.

I didn't switch the input to multiline, because it turns out the multiline component is broken in various ways, so I filed a separate story for that: DESKTOP-1527

Any other review comments?

**Comment by MarcoPolo** [comment]
👍

File: shared/profile/edit-profile/index.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/profile/edit-profile/index.js]: https://github.com/keybase/client/blob/master/shared/router/meta-navigator.js#L93

It expects the subroute to have a parseRoute, I was wrong when I said it didn't.

It uses it to get info for componentAtTop like the title, and optionally component.

In the code it gets the nextComponent correct from the parent in the route, but if this didn't have a parseRoute it wouldn't run the loop again and the component that would be renderered would be the parent and not the child in nextComponent.

**Pull request merged**